tests/translator/input/error_layer_invalid_properties.yaml

```diff
@@ -43,4 +46,10 @@ Resources:
     Type: 'AWS::Serverless::LayerVersion'
     Properties:
       ContentUri: s3://sam-demo-bucket/layer.zip
-      RetentionPolicy: !Ref DeleteList
\ No newline at end of file
+      RetentionPolicy: !Ref DeleteList
+
+  LayerWithRetentionPolicyAsIntrinsic:
+    Type: 'AWS::Serverless::LayerVersion'
+    Properties:
+      ContentUri: s3://sam-demo-bucket/layer.zip
+      RetentionPolicy: !If [DeletePolicyIf, Retain, Delete]
```

Comment:
Do we have a test for Ref? I think that intrinsic is supported right?

Reply:
Yeap it is supported, and we already have tests in `basic_layer.yaml` file above.